src/main/scala/org/zalando/spearheads/innkeeper/services/ServiceResult.scala

```diff
@@ -7,11 +7,20 @@ import org.zalando.spearheads.innkeeper.api.RouteOut
  */
 trait ServiceResult {
 
-  sealed trait Result[+T]
+  sealed trait Result[+T] {
 
-  case class Success[+T](result: T) extends Result[T]
+    def map[G](f: T => G): Result[G] = this match {
+      case Success(result) => Success(f(result))
+      case Failure(r)      => Failure(r)
```

Comment:
Isn't this needlessly creating a new Failure instance? why not just `case x => x` to reuse the old one?